core/runtime/src/main/java/io/quarkus/artemis/core/runtime/ArtemisCoreRecorder.java

```diff
@@ -16,17 +17,27 @@ public Supplier<ServerLocator> getServerLocatorSupplier(
         ArtemisRuntimeConfig runtimeConfig = runtimeConfigs.getAllConfigs().getOrDefault(name, new ArtemisRuntimeConfig());
         ArtemisBuildTimeConfig buildTimeConfig = buildTimeConfigs.getAllConfigs().getOrDefault(name,
                 new ArtemisBuildTimeConfig());
-        ArtemisUtil.validateIntegrity(runtimeConfig, buildTimeConfig, name);
+        ArtemisUtil.validateIntegrity(name, runtimeConfig, buildTimeConfig);
+        ServerLocator serverLocator = getServerLocator(runtimeConfig);
         return new Supplier<>() {
             @Override
             public ServerLocator get() {
-                try {
-                    return ActiveMQClient.createServerLocator(runtimeConfig.getUrl());
```

Comment:
Is there no side effect of creating the ServerLocater earlier than the call to `Supplier.get` ?

Reply:
Refactored to create the locator in the supplier. Moved the integrity check also down to the supplier.

Reply:
Refactored to create the locator in the supplier. Moved the integrity check also down to the supplier.